src/database/DataTypes/Requisition.js

```diff
@@ -61,6 +65,26 @@ export class Requisition extends Realm.Object {
     this.addItem(requisitionItem);
   }
 
+  createCustomerInvoice(database, user) {
+    if (this.isRequest || this.isFinalised) {
+      throw new Error('Cannot create invoice from Finalised or Request Requistion ');
+    }
+    if (database.objects('Transaction').
+            filtered('linkedRequisition.id = $0', this.id).length > 0) return;
```

Comment:
This is for sync, in case of resync and CustomerInvoice arrived before Requisition. And if CustomerInvoice arrived after Requisition.. this is why this should only be done in postSyncProcess